### PR TITLE
Dynamically adapt pretty-printing according to the terminal width

### PR DESCRIPTION
**Pull request by CraigFe**
This PR does two things:

- determines the terminal width dynamically, by adding a dependency on `terminal_size`. (This library currently depends on `Alcotest` for testing, so we may need to vendor it.)

- fixes pretty-printing of long test names, by truncating them with "`...`". This ensures that they never span more than one line, which previously caused trouble for carriage returns (fix https://github.com/mirage/alcotest/issues/72) and the unicode boxes (fix https://github.com/mirage/alcotest/issues/277).

### CODE REVIEW

**Comment by CraigFe** [comment]
Looks like `terminal_size` doesn't support Windows, so we'll need to replicate the stubs here (either with a default for Windows or using a [different solution](https://stackoverflow.com/questions/6812224/getting-terminal-size-in-c-for-windows)). @emillon: any thoughts on which approach to take?

**Comment by CraigFe** [comment]
I scrapped the idea of using `terminal_size` and wrote my own set of stubs which can support Windows + don't have the cyclic dependency issue.

This seems to pass CI on all tested platforms, so can be considered for merge.

**Comment by CraigFe** [comment]
Will merge this next week if there are no objections.

File: src/alcotest/alcotest_stubs.c [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	CAMLreturn(result);
+}
+
+// Unsupported platform
+#else
+
+CAMLprim value ocaml_alcotest_get_terminal_dimensions(value unit)
+{
+	CAMLparam1(unit);
+	CAMLlocal2(result, pair);
```
Comment (emillon) [inline src/alcotest/alcotest_stubs.c:76]: Not that it changes anything
```suggestion
	CAMLlocal1(result);
```

Comment (CraigFe) [reply src/alcotest/alcotest_stubs.c:76]: Indeed, thanks.

**Pull request merged**